spinalcordtoolbox/reports/qc2.py

```diff
@@ -984,3 +1021,54 @@ def assign_label_colors_by_groups(labels):
                 color_list[label - labels.min()] = sampled_colors[j % len(sampled_colors)]
 
     return color_list
+
+
+def crop_with_mask(array, img_crop, pad=3, axis=0, total_slices=None):
+    """
+    Crop array along a specific axis based on nonzero slices in the reference image.
+
+    Use `pad` if you want to pad around the mask (no matter how big the mask is). Use
+    `total_slices` to pad only until the total number of slices is reached (overrides `pad`).
+
+    Note: We use axis=0 by default because QC images are reoriented to SAL, therefore we
+        crop the SI axis by default.
+    """
+    # get extents of segmentation used for cropping
+    first_slice = min(np.where(img_crop.data)[axis])
+    last_slice = max(np.where(img_crop.data)[axis])
+    # if `total_slices` is specified, then override `pad`
+    if total_slices is not None:
+        pad_total = max(0, total_slices - (last_slice - first_slice + 1))
+        l_pad = math.floor(pad_total // 2)
+        r_pad = math.ceil(pad_total // 2)
+    # otherwise, use the provided value as-is
+    else:
+        l_pad = r_pad = pad
+    # pad (but make sure the indices are within the image bounds)
+    start = max(first_slice - l_pad, 0)
+    stop = min(last_slice + r_pad, img_crop.data.shape[axis])
+    # crop the image at the specified axis
+    idx = [slice(None)] * array.ndim   # Start with full image
+    idx[axis] = slice(start, stop + 1)  # Limit axis to slice range
+    return array[tuple(idx)]
+
+
+def get_max_radius(img, plane='Axial'):
+    """Determine the maximum slicewise width/height of the nonzero voxels in img."""
+    if plane == 'Axial':
+        # In Axial plane, the radius is the maximum width/height of the spinal cord mask dilated by 20% or 15, whichever is larger.
+        dilation = 1.2
+        default = 15
+        widths = [np.max(np.where(slice)[0]) - np.min(np.where(slice)[0]) if np.sum(slice) > 0 else 0 for slice in img.data]
+        heights = [np.max(np.where(slice)[1]) - np.min(np.where(slice)[1]) if np.sum(slice) > 0 else 0 for slice in img.data]
```

Comment:
I think the names `heights` and `widths` are switched? In the Axial case, SAL orientation, `np.where(slice)[0]` are coordinates in the AP direction, which is ultimately displayed vertically in the QC report. And `np.where(slice)[1]` are coordinates in the LR direction, which is displayed horizontally in the QC.

This would better align the return value with the Sagittal case, which returns `height, width` instead of `width, height`. (Or maybe this is a bigger structural problem between Axial and Sagittal mosaic code.)